include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -436,147 +440,215 @@ class alignment_algorithm :
-    * to the infix of the corresponding range and return the aligned sequence.
-    */
-    template <typename first_range_t, typename second_range_t>
-    auto compute_traceback(first_range_t & first_range,
-                           second_range_t & second_range,
-                           alignment_coordinate back_coordinate)
+    /*!\brief Creates a new alignment result from the current alignment optimum and for the given pair of ranges.
+     * \param[in] index The internal index used for this pair of sequences.
+     * \param[in] fst_range The first range to get the alignment for if requested.
+     * \param[in] snd_range The second range to get the alignment for if requested.
+     * \param[in] cache The current alignment cache containing the computed optimum.
+     *
+     * \returns A new alignment result with the requested values.
+     *
+     * \details
+     *
+     *
+     */
+    template <typename index_t, typename fst_range_t, typename snd_range_t, typename cache_t>
+    auto make_result(index_t index, fst_range_t & fst_range, snd_range_t & snd_range, cache_t const & cache)
     {
-        // Parse the traceback
-        auto [front_coordinate, first_gap_segments, second_gap_segments] = this->parse_traceback(back_coordinate);
+        static_assert(config_t::template exists<align_cfg::result>(),
+                "The configuration must contain an align_cfg::result element.");
 
-        using result_type = typename align_result_selector<first_range_t, second_range_t, config_t>::type;
-        using aligned_seq_type = decltype(result_type{}.alignment);
+        using result_t = typename align_result_selector<fst_range_t, snd_range_t, config_t>::type;
+        result_t res{};
 
-        // If we compute the traceback the aligned sequences must be provided.
-        if constexpr (seqan3::tuple_like<aligned_seq_type>)
-        {
-            auto fill_aligned_sequence = [] (auto & aligned_sequence_, auto & gap_segments, size_t const normalise)
-            {
-                assert(std::ranges::empty(gap_segments) || normalise <= gap_segments[0].position);
-
-                size_t offset = 0;
-                for (auto const & gap_elem : gap_segments)
-                {
-                    auto it = std::ranges::begin(aligned_sequence_);
-                    std::ranges::advance(it, (gap_elem.position - normalise) + offset);
-                    insert_gap(aligned_sequence_, it, gap_elem.size);
-                    offset += gap_elem.size;
-                }
-            };
+        res.id = index;
 
-            // In banded case we need to refine the back coordinate to map to the correct position within the
-            // second range.
+        using result_config_type = std::remove_reference_t<decltype(
+                seqan3::get<align_cfg::result>(std::declval<config_t>()).value)>;
+
+        // Choose what needs to be computed.
+        if constexpr (result_config_type::rank >= 0)  // compute score
```

Comment:
Because the information given needs to be static anyway, since it determines the result output which allows us to define a std::function object with certain dynamic configurations in place. Since the interface should be similar to the other config interfaces these are encoded as different types. To do the things you can do with enums I added the static constexpr variable as seena bove. but using an enum for this would be overkill I guess

Reply:
I really don't see the design gain here.

> Because the information given needs to be static anyway

An enum can be static?

> Since the interface should be similar to the other config interfaces

But those are much more complex and sometimes require extra parameters again.
You just need flags. Enums model flags. 

Now you have 4 structs with 4 member variables and 4 constexpr variables.
Instead of 1 scoped enum with 4 values.

Reply:
There is a larger design goal behind all this, that makes the use of the config arguments of algorithms for the users much easier.

Reply:
I'd love to explain this to you, but it is better done on offline.